### DISCUSSION TITLE
perspective-viewer-d3fc - "TypeError: Cannot read properties of undefined (reading 'indexOf')" on running jest tests

### DISCUSSION THREAD

**Discussion by RaviTewari1** [perspective-dev/perspective#2656] [Q&A]
Getting error while running jest test cases in react
Error on this line - import '@finos/perspective-viewer-d3fc';
Here is the call stack:
      at indexOf (../node_modules/@finos/perspective-viewer-d3fc/src/ts/plugin/plugin.ts:59:25)
          at Proxy.forEach (<anonymous>)
      at forEach (../node_modules/@finos/perspective-viewer-d3fc/src/ts/plugin/plugin.ts:58:18)
      at Object.bi (../node_modules/@finos/perspective-viewer-d3fc/src/ts/plugin/plugin.ts:36:34)
How can I resolve it. please suggest

**Comment by texodus** [comment]
You need to provide a repro if you want someone to help.

**Reply by RaviTewari1** [reply]
I am sorry if my question isnt detailed enough, I will try, let me know if anything specific is required.
This is roughly my test file
import { render} from '@testing-library/react';
it('should render ', async () => {
    const { getByText } = render(<MyPerspectiveComponent />); 
    });
In my component I am using all these libraries:
import perspective from '@finos/perspective';
import '@finos/perspective-viewer';
import '@finos/perspective-viewer-datagrid';
import '@finos/perspective-workspace';
import  '@finos/perspective-viewer-d3fc';  
    
When I run the jest test I get this error - "TypeError: Cannot read properties of undefined (reading 'indexOf')"
on this line - import  '@finos/perspective-viewer-d3fc';  
I face no issues running the code in browser, I can see the charts on the UI, issue only running test case.

**Reply by RaviTewari1** [reply]
How can I mock @finos/perspective-viewer-d3fc?

**Reply by texodus** [reply]
A repro is a GitHub project I can clone and run on its own, with instructions and dependencies, and which replicates the problem concisely.
